cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -122,15 +132,48 @@ int gpi_register_impl(GpiImplInterface *func_tbl)
     return 0;
 }
 
-void gpi_embed_init(gpi_sim_info_t *info)
+void gpi_register_sim_event_callback(void (*func)(void*, gpi_event_t, const char *), void* userdata)
 {
-    if (embed_sim_init(info))
+    sim_event_callbacks.push_back(std::make_pair(func, userdata));
+}
+
+void gpi_register_sim_end_callback(void (*func)(void*), void* userdata)
+{
+    sim_end_callbacks.push_back(std::make_pair(func, userdata));
+}
+
+static std::pair<std::string, std::string> parse_library_string(const char* begin, const char* end);
+static std::vector<std::pair<std::string, std::string>> parse_library_list_string(const char* lib_list);
+template <typename Func>
+static bool load_libraries(const std::vector<std::pair<std::string, std::string>>& libraries, Func&& load);
+
+void gpi_embed_init(int argc, char const* const* argv)
+{
+    // get GPI_USERS string
```

Comment:
This new envvar should be documented, and I would call it ``COCOTB_GPI_USERS``.

Reply:
GPI is a self-sufficient entity that lives in this repo. "cocotb" should be scrubbed from GPI code. I'm going to add sections to the environment variables tab to disambiguate the owner of a particular environment variables when I add documentation.

Reply:
Okay, GPI should then get its own documentation chapter (or "part") at some point (I'm  working on some general doc reorganization anyway). If GPI even moves to its own repo, we are prepared.